// @hitachivantara/uikit-react-core/dist/esm/Snackbar/Snackbar.js
import { jsx } from "@emotion/react/jsx-runtime";
import { useCallback } from "react";
import Slide from "@mui/material/Slide";
import Snackbar from "@mui/material/Snackbar";
import capitalize from "lodash/capitalize";
import { setId } from "../utils/setId.js";
import { useClasses } from "./Snackbar.styles.js";
import { staticClasses } from "./Snackbar.styles.js";
import { HvSnackbarContent } from "./SnackbarContent/SnackbarContent.js";
const HvSnackbar = ({
  classes: classesProp,
  className,
  id,
  open = false,
  onClose,
  label = "",
  anchorOrigin = { vertical: "top", horizontal: "right" },
  autoHideDuration = 5e3,
  variant = "default",
  showIcon = false,
  customIcon = null,
  action = null,
  actionCallback,
  transitionDuration = 300,
  transitionDirection = "left",
  container,
  offset = 60,
  snackbarContentProps,
  ...others
}) => {
  const { classes } = useClasses(classesProp);
  const anchorOriginOffset = {
    anchorOriginTop: {
      top: `${offset}px`
    },
    anchorOriginBottom: {
      bottom: `${offset}px`
    }
  };
  const SlideTransition = useCallback(
    (properties) => /* @__PURE__ */ jsx(
      Slide,
      {
        ...properties,
        container,
        direction: transitionDirection
      }
    ),
    [container, transitionDirection]
  );
  return /* @__PURE__ */ jsx(
    Snackbar,
    {
      style: anchorOriginOffset[`anchorOrigin${capitalize(anchorOrigin.vertical)}`],
      classes,
      className,
      id,
      anchorOrigin,
      open,
      onClose,
      autoHideDuration,
      transitionDuration,
      TransitionComponent: SlideTransition,
      ...others,
      children: /* @__PURE__ */ jsx(
        HvSnackbarContent,
        {
          id: setId(id, "content"),
          label,
          variant,
          customIcon,
          showIcon,
          action,
          actionCallback,
          ...snackbarContentProps
        }
      )
    }
  );
};
export {
  HvSnackbar,
  staticClasses as snackbarClasses
};
