// @hitachivantara/uikit-react-core/dist/cjs/TimeAgo/TimeAgo.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const useDefaultProps = require("../hooks/useDefaultProps.cjs");
const generic = require("../types/generic.cjs");
const TimeAgo_styles = require("./TimeAgo.styles.cjs");
const useTimeAgo = require("./useTimeAgo.cjs");
const Typography = require("../Typography/Typography.cjs");
const HvTimeAgo = generic.fixedForwardRef(function HvTimeAgo2(props, ref) {
  const {
    classes: classesProp,
    className,
    timestamp,
    locale: localeProp = "en",
    component: Component = Typography.HvTypography,
    emptyElement = "—",
    disableRefresh = false,
    showSeconds = false,
    justText = false,
    ...others
  } = useDefaultProps.useDefaultProps("HvTimeAgo", props);
  const { classes, cx } = TimeAgo_styles.useClasses(classesProp);
  const locale = localeProp || "en";
  const timeAgo = useTimeAgo.default(timestamp, {
    locale,
    disableRefresh,
    showSeconds
  });
  if (justText && timestamp)
    return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, { children: timeAgo });
  return /* @__PURE__ */ jsxRuntime.jsx(Component, { ref, className: cx(classes.root, className), ...others, children: !timestamp ? emptyElement : timeAgo });
});
exports.timeAgoClasses = TimeAgo_styles.staticClasses;
exports.HvTimeAgo = HvTimeAgo;
