// @hitachivantara/uikit-react-core/dist/cjs/Tooltip/Tooltip.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const Tooltip = require("@mui/material/Tooltip");
const Fade = require("@mui/material/Fade");
const useTheme = require("../hooks/useTheme.cjs");
const useDefaultProps = require("../hooks/useDefaultProps.cjs");
const document = require("../utils/document.cjs");
const Tooltip_styles = require("./Tooltip.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const Tooltip__default = /* @__PURE__ */ _interopDefault(Tooltip);
const Fade__default = /* @__PURE__ */ _interopDefault(Fade);
const HvTooltip = React.forwardRef((props, ref) => {
  const {
    className,
    classes: classesProp,
    open,
    enterDelay = 300,
    placement = "top",
    useSingle = true,
    children,
    title,
    TransitionComponent = Fade__default.default,
    TransitionProps = { timeout: 400, placement: "top" },
    containerId,
    ...others
  } = useDefaultProps.useDefaultProps("HvTooltip", props);
  const { rootId } = useTheme.useTheme();
  const { classes } = Tooltip_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tooltip__default.default,
    {
      ref,
      open: open ?? void 0,
      enterDelay,
      placement,
      TransitionComponent,
      TransitionProps,
      className,
      classes: {
        tooltip: useSingle ? classes.tooltip : classes.tooltipMulti,
        popper: classes.popper
      },
      title,
      PopperProps: {
        container: document.getElementById(containerId || rootId)
      },
      ...others,
      children
    }
  );
});
exports.tooltipClasses = Tooltip_styles.staticClasses;
exports.HvTooltip = HvTooltip;
