// @hitachivantara/uikit-react-core/dist/cjs/ScrollTo/Horizontal/ScrollToHorizontal.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const { staticClasses, useClasses } = classes.createClasses(
  "HvScrollToHorizontal",
  {
    root: {
      display: "flex",
      padding: "0 30px",
      listStyleType: "none",
      flexWrap: "wrap",
      backdropFilter: `blur(4px)`,
      backgroundColor: uikitStyles.theme.alpha("atmo2", 0.9)
    },
    positionSticky: {
      position: "sticky",
      zIndex: `calc(${uikitStyles.theme.zIndices.banner} - 2)`,
      top: 0,
      left: 0
    },
    positionFixed: {
      position: "fixed",
      zIndex: `calc(${uikitStyles.theme.zIndices.banner} - 2)`,
      top: 0,
      left: 0
    },
    notSelectedRoot: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      height: "16px",
      width: "16px",
      borderRadius: "50%"
    },
    notSelected: {
      height: "4px",
      width: "4px",
      borderRadius: "50%",
      display: "inline-block",
      backgroundColor: uikitStyles.theme.colors.secondary_60
    },
    selected: {
      display: "flex",
      height: "16px",
      width: "16px"
    }
  }
);
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
