// @hitachivantara/uikit-react-core/dist/cjs/VerticalNavigation/NavigationSlider/NavigationSlider.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const NavigationSlider_styles = require("./NavigationSlider.styles.cjs");
const Button = require("../../Button/Button.cjs");
const ListContainer = require("../../ListContainer/ListContainer.cjs");
const ListItem = require("../../ListContainer/ListItem/ListItem.cjs");
const OverflowTooltip = require("../../OverflowTooltip/OverflowTooltip.cjs");
const HvVerticalNavigationSlider = (props) => {
  const {
    id,
    classes: classesProp,
    data,
    selected,
    onNavigateToTarget,
    onNavigateToChild,
    forwardButtonAriaLabel = "Navigate to submenu"
  } = useDefaultProps.useDefaultProps("HvVerticalNavigationSlider", props);
  const { classes } = NavigationSlider_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(ListContainer.HvListContainer, { interactive: true, id, children: data && data.map((item) => /* @__PURE__ */ jsxRuntime.jsx(
    ListItem.HvListItem,
    {
      classes: {
        root: classes.root,
        selected: classes.listItemSelected,
        focus: classes.listItemFocus
      },
      onClick: (event) => {
        onNavigateToTarget?.(event, item);
      },
      "aria-label": item.label,
      "aria-current": selected === item.id ? item.href ? "page" : true : void 0,
      selected: selected === item.id,
      startAdornment: item.icon ? /* @__PURE__ */ jsxRuntime.jsx("div", { children: item.icon }) : void 0,
      endAdornment: item.data && item.data.length > 0 ? /* @__PURE__ */ jsxRuntime.jsx(
        Button.HvButton,
        {
          icon: true,
          onClick: (event) => {
            onNavigateToChild?.(event, item);
          },
          "aria-label": forwardButtonAriaLabel,
          children: /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.DropRightXS, {})
        }
      ) : void 0,
      children: /* @__PURE__ */ jsxRuntime.jsx(OverflowTooltip.HvOverflowTooltip, { data: item.label })
    },
    item.id
  )) });
};
exports.verticalNavigationSliderClasses = NavigationSlider_styles.staticClasses;
exports.HvVerticalNavigationSlider = HvVerticalNavigationSlider;
