// @hitachivantara/uikit-react-core/dist/esm/Tag/Tag.js
import { jsx } from "@emotion/react/jsx-runtime";
import { forwardRef } from "react";
import { getColor } from "@hitachivantara/uikit-styles";
import Chip from "@mui/material/Chip";
import { CloseXS, CheckboxCheck, Checkbox } from "@hitachivantara/uikit-react-icons";
import { useTheme } from "../hooks/useTheme.js";
import { useDefaultProps } from "../hooks/useDefaultProps.js";
import { useControlled } from "../hooks/useControlled.js";
import { useClasses } from "./Tag.styles.js";
import { staticClasses } from "./Tag.styles.js";
const getCategoricalColor = (customColor, colors) => {
  return customColor && colors?.[customColor] || customColor || colors?.cat1;
};
const HvTag = forwardRef((props, ref) => {
  const {
    classes: classesProp,
    className,
    style,
    label,
    disabled,
    type = "semantic",
    selectable,
    selected,
    defaultSelected = false,
    color,
    deleteIcon,
    onDelete,
    onClick,
    // TODO: remove from API
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    deleteButtonArialLabel = "Delete tag",
    deleteButtonProps = {},
    ...others
  } = useDefaultProps("HvTag", props);
  const { colors } = useTheme();
  const { classes, cx, css } = useClasses(classesProp);
  const [isSelected, setIsSelected] = useControlled(
    selected,
    Boolean(defaultSelected)
  );
  const defaultDeleteIcon = /* @__PURE__ */ jsx(
    CloseXS,
    {
      role: "none",
      className: cx(classes.button, classes.tagButton),
      iconSize: "XS",
      ...deleteButtonProps
    }
  );
  const categoricalBackgroundColor = type === "categorical" ? getCategoricalColor(color, colors) : void 0;
  const backgroundColor = type === "semantic" && getColor(color, "neutral_20") || type === "categorical" && `${categoricalBackgroundColor}30` || void 0;
  const isClickable = !!(onClick || onDelete) && !disabled;
  const clickableClass = css({
    "&:hover": {
      boxShadow: `0 0 0 1pt ${categoricalBackgroundColor}`
    }
  });
  const colorOverride = disabled && ["atmo3", "secondary_60"] || void 0;
  const avatarIcon = isSelected ? /* @__PURE__ */ jsx(CheckboxCheck, { color: colorOverride, iconSize: "XS" }) : /* @__PURE__ */ jsx(Checkbox, { color: colorOverride, iconSize: "XS" });
  return /* @__PURE__ */ jsx(
    Chip,
    {
      ref,
      label,
      disabled,
      className: cx({ [clickableClass]: isClickable }, className),
      style: {
        ...disabled ? null : { backgroundColor },
        ...style
      },
      classes: {
        root: cx(classes.root, classes.chipRoot, {
          [classes.disabled]: disabled,
          [classes.clickable]: isClickable,
          [classes.categorical]: type === "categorical",
          [classes.categoricalFocus]: type === "categorical" && !disabled,
          [classes.categoricalDisabled]: type === "categorical" && disabled
        }),
        label: classes.label,
        deleteIcon: cx(classes.deleteIcon, {
          [classes.disabledDeleteIcon]: disabled
        })
      },
      deleteIcon: deleteIcon || defaultDeleteIcon,
      onDelete: disabled ? void 0 : onDelete,
      onClick: (event) => {
        if (disabled)
          return;
        if (selectable)
          setIsSelected(!isSelected);
        onClick?.(event, !isSelected);
      },
      "aria-pressed": isSelected,
      ...selectable && type === "semantic" && {
        avatar: avatarIcon
      },
      ...others
    }
  );
});
export {
  HvTag,
  staticClasses as tagClasses
};
