// @hitachivantara/uikit-react-core/dist/cjs/ScrollTo/Horizontal/HorizontalScrollListItem/HorizontalScrollListItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../utils/focusUtils.cjs");
const classes = require("../../../utils/classes.cjs");
const name = "HvHorizontalScrollListItem";
const { staticClasses, useClasses } = classes.createClasses(name, {
  root: {
    padding: "10px 0"
  },
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "48px",
    cursor: "pointer",
    borderBottom: "none",
    "&:hover": {
      backgroundColor: "transparent",
      "& $notSelected": {
        height: "6px",
        width: "6px",
        backgroundColor: uikitStyles.theme.colors.secondary
      },
      "& $notSelectedRoot": {
        backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
      }
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    }
  },
  text: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    height: "48px",
    borderBottom: "none",
    "& p": {
      padding: "3px 10px",
      maxWidth: "120px",
      textOverflow: "ellipsis",
      overflow: "hidden"
    }
  },
  selected: {
    borderBottom: "none"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
