// @hitachivantara/uikit-react-core/dist/cjs/Tabs/Tabs.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const Tabs = require("@mui/material/Tabs");
const useDefaultProps = require("../hooks/useDefaultProps.cjs");
const Tabs_styles = require("./Tabs.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const Tabs__default = /* @__PURE__ */ _interopDefault(Tabs);
const HvTabs = (props) => {
  const { classes: classesProp, ...others } = useDefaultProps.useDefaultProps("HvTabs", props);
  const { classes } = Tabs_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tabs__default.default,
    {
      classes: {
        root: classes.root,
        flexContainer: classes.flexContainer,
        indicator: classes.indicator,
        scroller: classes.scroller
      },
      TabIndicatorProps: { children: /* @__PURE__ */ jsxRuntime.jsx("div", {}) },
      ...others
    }
  );
};
exports.tabsClasses = Tabs_styles.staticClasses;
exports.HvTabs = HvTabs;
