// @hitachivantara/uikit-react-core/dist/esm/Section/Section.js
import { jsxs, jsx } from "@emotion/react/jsx-runtime";
import { forwardRef } from "react";
import { Up, Down } from "@hitachivantara/uikit-react-icons";
import { useDefaultProps } from "../hooks/useDefaultProps.js";
import { useControlled } from "../hooks/useControlled.js";
import { useUniqueId } from "../hooks/useUniqueId.js";
import { setId } from "../utils/setId.js";
import { useClasses } from "./Section.styles.js";
import { staticClasses } from "./Section.styles.js";
import { HvButton } from "../Button/Button.js";
const HvSection = forwardRef(
  (props, ref) => {
    const {
      id,
      classes: classesProp,
      className,
      title,
      expandable,
      expanded,
      defaultExpanded = true,
      actions,
      onToggle,
      expandButtonProps,
      raisedHeader,
      contentRef,
      children,
      ...others
    } = useDefaultProps("HvSection", props);
    const { classes, cx } = useClasses(classesProp);
    const [isOpen, setIsOpen] = useControlled(
      expanded,
      Boolean(defaultExpanded)
    );
    const elementId = useUniqueId(id, "hvSection");
    const contentId = setId(elementId, "content");
    const showContent = expandable ? !!isOpen : true;
    return /* @__PURE__ */ jsxs(
      "div",
      {
        ref,
        id: elementId,
        className: cx(classes.root, className),
        ...others,
        children: [
          (title || actions || expandable) && /* @__PURE__ */ jsxs(
            "div",
            {
              className: cx(classes.header, {
                [classes.raisedHeader]: raisedHeader && isOpen
              }),
              children: [
                expandable && /* @__PURE__ */ jsx(
                  HvButton,
                  {
                    icon: true,
                    onClick: (event) => {
                      setIsOpen((o) => !o);
                      onToggle?.(event, !isOpen);
                    },
                    "aria-expanded": isOpen,
                    "aria-controls": contentId,
                    "aria-label": isOpen ? "Collapse" : "Expand",
                    ...expandButtonProps,
                    children: isOpen ? /* @__PURE__ */ jsx(Up, {}) : /* @__PURE__ */ jsx(Down, {})
                  }
                ),
                title,
                /* @__PURE__ */ jsx("div", { className: classes.actions, children: actions })
              ]
            }
          ),
          /* @__PURE__ */ jsx(
            "div",
            {
              ref: contentRef,
              id: contentId,
              hidden: !isOpen,
              className: cx(classes.content, {
                [classes.hidden]: !showContent,
                [classes.spaceTop]: !(title || actions || expandable)
              }),
              children
            }
          )
        ]
      }
    );
  }
);
export {
  HvSection,
  staticClasses as sectionClasses
};
