// @hitachivantara/uikit-react-core/dist/cjs/Select/OptionGroup.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const OptionGroup = require("@mui/base/OptionGroup");
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../utils/classes.cjs");
const useDefaultProps = require("../hooks/useDefaultProps.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvOptionGroup", {
  root: {
    listStyle: "none",
    ...uikitStyles.theme.typography.label
  }
});
const HvOptionGroup = React.forwardRef(
  (props, ref) => {
    const {
      className,
      classes: classesProp,
      ...others
    } = useDefaultProps.useDefaultProps("HvOptionGroup", props);
    const { classes: classes2, cx } = useClasses(classesProp);
    return /* @__PURE__ */ jsxRuntime.jsx(
      OptionGroup.OptionGroup,
      {
        ref,
        className: cx(classes2.root, className),
        ...others
      }
    );
  }
);
exports.HvOptionGroup = HvOptionGroup;
exports.optionGroupClasses = staticClasses;
