// @hitachivantara/uikit-react-core/dist/cjs/Select/Select.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../utils/classes.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvSelect", {
  root: {
    position: "relative",
    "&$disabled,&$readOnly": {
      pointerEvents: "none"
    }
  },
  disabled: {},
  readOnly: {},
  invalid: {
    border: `1px solid ${uikitStyles.theme.colors.negative}`
  },
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {
    display: "block",
    paddingBottom: 6
  },
  description: {},
  select: {},
  popper: {
    zIndex: uikitStyles.theme.zIndices.popover
  },
  panel: {
    border: `1px solid ${uikitStyles.theme.colors.secondary}`,
    marginTop: -1,
    marginBottom: -1
  },
  panelOpenedUp: {
    borderRadius: `${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} 0 0`
  },
  panelOpenedDown: {
    borderRadius: `0 0 ${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base}`
  },
  error: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
