// @hitachivantara/uikit-react-core/dist/cjs/Select/SelectButton.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const uikitStyles = require("@hitachivantara/uikit-styles");
const useDefaultProps = require("../hooks/useDefaultProps.cjs");
const classes = require("../utils/classes.cjs");
const Button = require("../Button/Button.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvSelectButton", {
  root: {
    width: "100%",
    minWidth: "unset",
    userSelect: "none",
    position: "relative",
    paddingLeft: uikitStyles.theme.space.xs,
    justifyContent: "flex-start"
  },
  disabled: {},
  readOnly: {
    userSelect: "text",
    backgroundColor: uikitStyles.theme.colors.atmo2
  },
  open: {
    backgroundColor: uikitStyles.theme.colors.atmo1
  },
  openUp: {
    borderRadius: `0px 0px ${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base}`
  },
  openDown: {
    borderRadius: `${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} 0px 0px`
  },
  selection: {
    color: "inherit",
    flex: 1,
    textAlign: "start",
    overflow: "clip visible",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  },
  placeholder: {},
  arrowContainer: {
    marginRight: uikitStyles.theme.spacing(-2)
  },
  arrow: {
    transition: "rotate 0.2s ease"
  }
});
const HvSelectButton = React.forwardRef((props, ref) => {
  const {
    classes: classesProp,
    className,
    open,
    placement = "bottom",
    disabled,
    readOnly,
    children,
    adornment,
    ...others
  } = useDefaultProps.useDefaultProps("HvSelectButton", props);
  const { classes: classes2, cx } = useClasses(classesProp);
  const endIcon = adornment ?? /* @__PURE__ */ jsxRuntime.jsx(
    uikitReactIcons.DropDownXS,
    {
      iconSize: "XS",
      style: { rotate: open ? "180deg" : void 0 },
      className: classes2.arrow
    }
  );
  return /* @__PURE__ */ jsxRuntime.jsx(
    Button.HvButton,
    {
      ref,
      variant: "secondarySubtle",
      disabled: disabled || readOnly,
      className: cx(classes2.root, className, {
        [classes2.open]: open,
        [classes2.openUp]: open && placement.includes("top"),
        [classes2.openDown]: open && placement.includes("bottom"),
        [classes2.disabled]: disabled,
        [classes2.readOnly]: readOnly
      }),
      classes: { endIcon: classes2.arrowContainer },
      endIcon,
      ...others,
      children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes2.selection, children: children && typeof children === "string" ? /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes2.placeholder, children }) : children })
    }
  );
});
exports.HvSelectButton = HvSelectButton;
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
