// @hitachivantara/uikit-react-core/dist/cjs/Calendar/SingleCalendar/SingleCalendar.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvSingleCalendar", {
  calendarContainer: {
    backgroundColor: uikitStyles.theme.colors.atmo1,
    width: "318px",
    minHeight: "440px",
    position: "relative"
  },
  calendarWrapper: {
    overflow: "hidden",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    padding: uikitStyles.theme.spacing("sm")
  },
  calendarGrid: {
    display: "flex",
    flexFlow: "wrap",
    width: "280px",
    "& $cellsInRange": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      "& $startBookend": {
        borderLeft: `1px solid ${uikitStyles.theme.colors.secondary}`,
        backgroundColor: uikitStyles.theme.colors.atmo3,
        "&:hover": {
          borderRight: `1px solid ${uikitStyles.theme.colors.secondary}`
        }
      }
    },
    "&:hover $cellsInRange": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      "& $startBookend": {
        borderLeft: `1px solid ${uikitStyles.theme.colors.secondary}`,
        borderRight: "inherit"
      }
    },
    "& $cellsInRange:hover": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      "& $calendarDate": {
        borderRight: `1px solid ${uikitStyles.theme.colors.secondary}`
      }
    },
    "& $cellsInRange:hover ~ $cellsInRange": {
      backgroundColor: uikitStyles.theme.colors.atmo1
      // controls the right side of the hovered range
    },
    "& $cellsOutsideRange:hover ~ $cellsInRange": {
      backgroundColor: uikitStyles.theme.colors.atmo1
      // control the right side when hovering outside of the range
    }
  },
  calendarDay: {
    display: "flex",
    justifyContent: "center",
    flexDirection: "column",
    textAlign: "center",
    height: "40px",
    width: "40px"
  },
  navigationContainer: {},
  focusSelection: {},
  navigationMonth: {},
  calendarDate: {},
  calendarDateNotInMonth: {},
  calendarDateSelected: {},
  calendarDateInvalid: {},
  calendarDateInSelectionRange: {},
  startBookend: {},
  endBookend: {},
  cellsInRange: {
    backgroundColor: uikitStyles.theme.colors.atmo3
  },
  cellsOutsideRange: {},
  cellContainer: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
