// @hitachivantara/uikit-react-core/dist/cjs/ColorPicker/SavedColors/SavedColors.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const name = "HvColorPicker-SavedColors";
const { staticClasses, useClasses } = classes.createClasses(name, {
  addButton: {
    margin: "4px",
    width: "32px",
    height: "32px"
  },
  root: {
    display: "flex",
    flexWrap: "wrap",
    position: "relative",
    alignItems: "center",
    padding: 0,
    width: "calc(100% + 8px)",
    margin: "-4px -4px"
  },
  swatchRoot: { position: "relative" },
  swatchWrap: {
    width: "32px",
    height: "32px",
    margin: "4px",
    "& > span > div": {
      borderRadius: uikitStyles.theme.radii.base
    }
  },
  removeButtonRoot: {
    position: "absolute",
    top: 0,
    right: 0,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: 16,
    width: 16,
    background: uikitStyles.theme.colors.atmo1,
    borderRadius: uikitStyles.theme.radii.base
  },
  removeButton: {
    height: 16,
    width: 16,
    minWidth: 16,
    minHeight: 16,
    padding: 0,
    margin: 0,
    "& div > span": {
      height: 16,
      width: 16,
      "& > div": { height: 16, width: 16 }
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
