// @hitachivantara/uikit-react-core/dist/cjs/InlineEditor/InlineEditor.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../utils/classes.cjs");
const BaseInput_styles = require("../BaseInput/BaseInput.styles.cjs");
require("../BaseInput/BaseInput.cjs");
const Input_styles = require("../Input/Input.styles.cjs");
require("../Input/Input.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvInlineEditor", {
  root: {
    [`& .${BaseInput_styles.staticClasses.inputRoot}.${Input_styles.staticClasses.inputRoot}`]: {
      height: "100%",
      minHeight: "32px"
    }
  },
  inputBorderContainer: {
    top: "unset",
    bottom: 0
  },
  input: {},
  inputRoot: {},
  text: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    alignSelf: "center"
  },
  largeText: {},
  textEmpty: {
    color: uikitStyles.theme.colors.secondary_60
  },
  button: {
    padding: uikitStyles.theme.spacing("6px", "8px", "5px", "8px"),
    minHeight: "32px",
    boxSizing: "border-box",
    cursor: "text",
    height: "100%",
    width: "100%",
    maxWidth: "100%",
    justifyContent: "flex-start",
    alignItems: "center",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    border: `1px solid transparent`,
    "&:hover, &:focus": {
      border: `1px solid ${uikitStyles.theme.colors.primary}`,
      backgroundColor: uikitStyles.theme.colors.atmo1,
      "& $icon": {
        visibility: "visible"
      }
    },
    "&:active": {
      border: `1px solid ${uikitStyles.theme.colors.secondary}`,
      backgroundColor: "transparent",
      "& $icon": {
        visibility: "visible"
      }
    },
    "& > div": {
      width: "100%"
    },
    "& > div > span": {
      width: "100%"
    }
  },
  icon: {
    cursor: "pointer",
    visibility: "hidden",
    alignSelf: "center",
    height: "16px",
    width: "32px",
    minWidth: "32px",
    "& svg": {
      margin: uikitStyles.theme.spacing(0, "xs")
    }
  },
  iconVisible: {
    visibility: "visible"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
