// @hitachivantara/uikit-react-core/dist/cjs/BreadCrumb/Page/Page.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const Page_styles = require("./Page.styles.cjs");
const Typography = require("../../Typography/Typography.cjs");
const OverflowTooltip = require("../../OverflowTooltip/OverflowTooltip.cjs");
const HvBreadCrumbPage = (props) => {
  const {
    component,
    onClick,
    elem,
    classes: classesProp
  } = useDefaultProps.useDefaultProps("HvBreadCrumbPage", props);
  const { classes, cx } = Page_styles.useClasses(classesProp);
  const { label, path, ...others } = elem;
  const handleClick = (event) => {
    event.preventDefault();
    onClick?.(event, elem);
  };
  return /* @__PURE__ */ jsxRuntime.jsx(
    Typography.HvTypography,
    {
      noWrap: true,
      variant: "label",
      component: component || "a",
      href: elem.path,
      onClick: onClick && handleClick,
      className: cx(classes.link, classes.label, classes.a),
      ...others,
      children: /* @__PURE__ */ jsxRuntime.jsx(OverflowTooltip.HvOverflowTooltip, { data: elem.label })
    }
  );
};
exports.breadCrumbPageClasses = Page_styles.staticClasses;
exports.HvBreadCrumbPage = HvBreadCrumbPage;
