// @hitachivantara/uikit-react-core/dist/cjs/BulkActions/BulkActions.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../utils/classes.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvBulkActions", {
  root: {
    display: "flex",
    alignItems: "center",
    border: `1px solid ${uikitStyles.theme.colors.atmo4}`,
    backgroundColor: uikitStyles.theme.colors.atmo2,
    padding: uikitStyles.theme.spacing("xs", "md"),
    marginBottom: uikitStyles.theme.space.xs
  },
  semantic: {
    backgroundColor: uikitStyles.theme.colors.containerBackgroundHover,
    "& $selectAll div": {
      color: "inherit",
      "&:hover:not(:disabled)": {
        backgroundColor: uikitStyles.theme.alpha("base_light", 0.3)
      },
      "& *": {
        color: "inherit",
        backgroundColor: "transparent"
      }
    },
    "& $selectAll:focus-within div": {
      backgroundColor: uikitStyles.theme.alpha("base_light", 0.3)
    }
  },
  actions: { display: "inline-flex", marginLeft: "auto" },
  selectAllContainer: { display: "flex", alignItems: "center" },
  selectAll: {},
  selectAllPages: {},
  divider: {
    display: "flex",
    backgroundColor: uikitStyles.theme.colors.atmo4,
    width: "1px",
    height: "32px",
    marginLeft: uikitStyles.theme.space.sm
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
