// @hitachivantara/uikit-react-core/dist/esm/TableSection/TableSection.styles.js
import { theme } from "@hitachivantara/uikit-styles";
import { createClasses } from "../utils/classes.js";
import { staticClasses as staticClasses$1 } from "../Table/TableContainer/TableContainer.styles.js";
import "../Table/TableContainer/TableContainer.js";
import { staticClasses as staticClasses$2 } from "../Table/TableHeader/TableHeader.styles.js";
import "../Table/TableHeader/TableHeader.js";
import { staticClasses as staticClasses$3 } from "../Table/TableRow/TableRow.styles.js";
import "../Table/TableRow/TableRow.js";
import { staticClasses as staticClasses$4 } from "../Table/TableCell/TableCell.styles.js";
import "../Table/TableCell/TableCell.js";
import { staticClasses as staticClasses$5 } from "../BulkActions/BulkActions.styles.js";
import "@emotion/react/jsx-runtime";
import { staticClasses as staticClasses$6 } from "../Pagination/Pagination.styles.js";
import "react";
import "@mui/material/Hidden";
import "@hitachivantara/uikit-react-icons";
const { staticClasses, useClasses } = createClasses("HvTableSection", {
  root: {},
  header: {
    // Only apply the border to divide the header and content when both are displayed
    "+ div": { borderTop: `1px solid ${theme.colors.atmo3}` }
  },
  actions: {
    right: theme.space.sm
  },
  content: {
    marginTop: 0,
    padding: 0,
    // Apply border radius to the first child if there's not an header
    "&:first-of-type": {
      "& > :first-of-type": {
        borderTopLeftRadius: theme.radii.round,
        borderTopRightRadius: theme.radii.round
      }
    },
    "& > :last-child": {
      borderBottomLeftRadius: theme.radii.round,
      borderBottomRightRadius: theme.radii.round
    },
    [`& .${staticClasses$1.root}`]: {
      paddingBottom: 0
    },
    [`& .${staticClasses$2.head}`]: {
      backgroundColor: theme.colors.atmo2,
      borderBottomColor: theme.colors.atmo3,
      [`&.${staticClasses$2.variantCheckbox}`]: {
        borderRight: "none"
      },
      [`&.${staticClasses$2.variantActions}`]: {
        borderLeft: "none"
      }
    },
    // Remove border for the last table row
    [`& .${staticClasses$3.root}`]: {
      "&:last-child": {
        [`& .${staticClasses$4.root}`]: { borderBottom: "none" }
      }
    },
    [`& .${staticClasses$4.root}`]: {
      borderBottomColor: theme.colors.atmo3
    },
    [`& .${staticClasses$4.variantCheckbox}`]: {
      borderRight: "none"
    },
    [`& .${staticClasses$4.variantActions}`]: {
      borderLeft: "none"
    },
    [`& .${staticClasses$5.root}`]: {
      marginBottom: 0,
      border: "none",
      borderBottom: `1px solid ${theme.colors.atmo3}`,
      padding: theme.spacing("xs", "sm")
    },
    [`& .${staticClasses$6.root}`]: {
      margin: 0,
      backgroundColor: theme.colors.atmo2,
      padding: theme.space.xs,
      borderTop: `1px solid ${theme.colors.atmo3}`
    },
    [`& .${staticClasses$6.pageSizeOptions}`]: {
      left: theme.space.sm
    },
    [`& .${staticClasses$6.pageSizeHeader}`]: {
      border: "none",
      "&:hover": {
        border: "none"
      }
    }
  },
  hidden: {},
  raisedHeader: {
    "+ div": {
      paddingTop: 0
    }
  },
  spaceTop: { paddingTop: 0 }
});
export {
  staticClasses,
  useClasses
};
