// @hitachivantara/uikit-react-core/dist/esm/ScrollTo/Horizontal/ScrollToHorizontal.js
import { jsx, jsxs } from "@emotion/react/jsx-runtime";
import { useMemo, useCallback } from "react";
import { useTheme } from "@mui/material/styles";
import useMediaQuery from "@mui/material/useMediaQuery";
import { theme } from "@hitachivantara/uikit-styles";
import { CurrentStep } from "@hitachivantara/uikit-react-icons";
import { useDefaultProps } from "../../hooks/useDefaultProps.js";
import { useUniqueId } from "../../hooks/useUniqueId.js";
import { useTheme as useTheme$1 } from "../../hooks/useTheme.js";
import { isKey } from "../../utils/keyboardUtils.js";
import { setId } from "../../utils/setId.js";
import { withTooltip } from "../withTooltip.js";
import { useScrollTo } from "../useScrollTo.js";
import { useClasses } from "./ScrollToHorizontal.styles.js";
import { staticClasses } from "./ScrollToHorizontal.styles.js";
import { HvHorizontalScrollListItem } from "./HorizontalScrollListItem/HorizontalScrollListItem.js";
const HvScrollToHorizontal = (props) => {
  const {
    id,
    defaultSelectedIndex = 0,
    scrollElementId,
    href = true,
    navigationMode = href ? "push" : "none",
    relativeLinks = false,
    onChange,
    onClick,
    onEnter,
    className,
    classes: classesProp,
    options,
    offset = 0,
    position = "relative",
    tooltipPosition = "top",
    ...others
  } = useDefaultProps("HvScrollToHorizontal", props);
  const { classes, css, cx } = useClasses(classesProp);
  const muiTheme = useTheme();
  const downSm = useMediaQuery(muiTheme.breakpoints.down("sm"));
  const upMd = useMediaQuery(muiTheme.breakpoints.up("md"));
  const { activeTheme } = useTheme$1();
  const elementId = useUniqueId(id, "hvHorizontalScrollto");
  const [selectedIndex, setScrollTo, elements] = useScrollTo(
    defaultSelectedIndex,
    scrollElementId,
    navigationMode,
    relativeLinks,
    offset,
    options,
    onChange
  );
  const handleSelection = (event, value, index) => {
    event.preventDefault();
    const wrappedOnChange = () => {
      onChange?.(event, index);
    };
    setScrollTo(event, value, index, wrappedOnChange);
  };
  const tooltipWrappers = useMemo(() => {
    return options.map((option) => {
      return withTooltip(option.label, "div", tooltipPosition);
    });
  }, [options, tooltipPosition]);
  const NotSelected = useCallback(() => {
    return /* @__PURE__ */ jsx("div", { className: classes.notSelectedRoot, children: /* @__PURE__ */ jsx("div", { className: classes.notSelected }) });
  }, [classes.notSelectedRoot, classes.notSelected]);
  const Selected = useCallback(() => {
    return /* @__PURE__ */ jsx(
      CurrentStep,
      {
        height: activeTheme?.scrollTo.dotSelectedSize,
        width: activeTheme?.scrollTo.dotSelectedSize,
        className: classes.selected
      }
    );
  }, [classes.selected, activeTheme?.scrollTo.dotSelectedSize]);
  const tabs = elements.map((option, index) => {
    const selected = selectedIndex === index;
    const tooltipWrapper = tooltipWrappers[index];
    return /* @__PURE__ */ jsxs(
      HvHorizontalScrollListItem,
      {
        id: setId(elementId, `item-${index}`),
        onClick: (event) => {
          if (navigationMode !== "none") {
            event.preventDefault();
          }
          handleSelection(event, option.value, index);
          onClick?.(event, index);
        },
        onKeyDown: (event) => {
          if (isKey(event, "Enter") === true) {
            if (navigationMode !== "none") {
              event.preventDefault();
            }
            handleSelection(event, option.value, index);
            onEnter?.(event, index);
          }
        },
        href: navigationMode !== "none" ? option.href : void 0,
        tooltipWrapper,
        selected,
        children: [
          /* @__PURE__ */ jsx("p", { children: option.label }),
          selected ? /* @__PURE__ */ jsx(Selected, {}) : /* @__PURE__ */ jsx(NotSelected, {})
        ]
      },
      option.key || option.label
    );
  });
  return /* @__PURE__ */ jsx(
    "ol",
    {
      className: cx(
        css({
          width: position === "fixed" && (upMd || downSm) ? `calc(100% - 2*${theme.spacing(upMd ? 4 : 2)})` : "100%",
          marginTop: 0,
          marginBottom: 0,
          marginRight: position === "fixed" && (upMd || downSm) ? theme.spacing(upMd ? 4 : 2) : 0,
          marginLeft: position === "fixed" && (upMd || downSm) ? theme.spacing(upMd ? 4 : 2) : 0
        }),
        classes.root,
        {
          [classes.positionSticky]: position === "sticky",
          [classes.positionFixed]: position === "fixed"
        },
        className
      ),
      id: elementId,
      ...others,
      children: tabs
    }
  );
};
export {
  HvScrollToHorizontal,
  staticClasses as scrollToHorizontalClasses
};
