// @hitachivantara/uikit-react-core/dist/cjs/Tab/Tab.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const Tab = require("@mui/material/Tab");
const useDefaultProps = require("../hooks/useDefaultProps.cjs");
const Tab_styles = require("./Tab.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const Tab__default = /* @__PURE__ */ _interopDefault(Tab);
const HvTab = (props) => {
  const {
    classes: classesProp,
    iconPosition = "top",
    disabled = false,
    ...others
  } = useDefaultProps.useDefaultProps("HvTab", props);
  const { classes, cx } = Tab_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tab__default.default,
    {
      classes: {
        root: classes.root,
        selected: classes.selected,
        disabled: classes.disabled
      },
      disableRipple: true,
      disableTouchRipple: true,
      focusVisibleClassName: cx("HvIsFocusVisible", classes.focusVisible),
      disabled,
      iconPosition,
      ...others
    }
  );
};
exports.tabClasses = Tab_styles.staticClasses;
exports.HvTab = HvTab;
