// @hitachivantara/uikit-react-core/dist/cjs/ActionsGeneric/ActionsGeneric.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const useDefaultProps = require("../hooks/useDefaultProps.cjs");
const setId = require("../utils/setId.cjs");
const ActionsGeneric_styles = require("./ActionsGeneric.styles.cjs");
const Button = require("../Button/Button.cjs");
const DropDownMenu = require("../DropDownMenu/DropDownMenu.cjs");
const HvActionsGeneric = (props) => {
  const {
    id,
    classes: classesProp,
    className,
    category = "secondaryGhost",
    disabled = false,
    actions = [],
    actionsCallback,
    maxVisibleActions = Infinity,
    ...others
  } = useDefaultProps.useDefaultProps("HvActionsGeneric", props);
  const { classes, cx } = ActionsGeneric_styles.useClasses(classesProp);
  if (!Array.isArray(actions))
    return React.isValidElement(actions) ? actions : null;
  const renderButton = (action, idx) => {
    const { disabled: actDisabled, id: actId, icon, label, ...other } = action;
    const actionId = setId.setId(id, idx, "action", action.id);
    const renderedIcon = React.isValidElement(icon) ? icon : icon?.({ isDisabled: disabled });
    return /* @__PURE__ */ jsxRuntime.jsx(
      Button.HvButton,
      {
        id: actionId,
        variant: category,
        className: classes.button,
        disabled: actDisabled ?? disabled,
        onClick: (event) => actionsCallback?.(event, id || "", action),
        startIcon: renderedIcon,
        ...other,
        children: label
      },
      actionId || idx
    );
  };
  const renderActionsGrid = () => {
    const actsVisible = actions.slice(0, maxVisibleActions);
    const actsDropdown = actions.slice(maxVisibleActions);
    const semantic = category === "semantic";
    const iconColor = disabled && "secondary_60" || semantic && "base_dark" || void 0;
    return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
      actsVisible.map((action, idx) => renderButton(action, idx)),
      /* @__PURE__ */ jsxRuntime.jsx(
        DropDownMenu.HvDropDownMenu,
        {
          id: setId.setId(id, "menu"),
          disabled,
          category,
          classes: {
            root: classes.dropDownMenu,
            icon: classes.dropDownMenuButton,
            iconSelected: classes.dropDownMenuButtonSelected
          },
          icon: /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.MoreOptionsVertical, { color: iconColor }),
          placement: "left",
          onClick: (event, action) => actionsCallback?.(event, id || "", action),
          dataList: actsDropdown,
          keepOpened: false,
          disablePortal: false
        }
      )
    ] });
  };
  const actionOverflow = actions.length > maxVisibleActions;
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cx(
        classes.root,
        { [classes.actionContainer]: actionOverflow },
        className
      ),
      ...others,
      children: actionOverflow ? renderActionsGrid() : actions.map((action, idx) => renderButton(action, idx))
    }
  );
};
exports.actionsGenericClasses = ActionsGeneric_styles.staticClasses;
exports.HvActionsGeneric = HvActionsGeneric;
