// @hitachivantara/uikit-react-core/dist/cjs/Forms/Label/Label.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const setId = require("../../utils/setId.cjs");
const FormUtils = require("../FormElement/utils/FormUtils.cjs");
const Label_styles = require("./Label.styles.cjs");
const FormElementContext = require("../FormElement/context/FormElementContext.cjs");
const Typography = require("../../Typography/Typography.cjs");
const HvLabel = (props) => {
  const {
    id,
    classes: classesProp,
    className,
    children,
    label,
    disabled,
    required,
    htmlFor: htmlForProp,
    ...others
  } = useDefaultProps.useDefaultProps("HvLabel", props);
  const { classes, cx } = Label_styles.useClasses(classesProp);
  const { elementId, elementDisabled, elementRequired } = React.useContext(FormElementContext.HvFormElementContext);
  const localDisabled = disabled || elementDisabled;
  const localRequired = required || elementRequired;
  const localId = id ?? setId.setId(elementId, "label");
  const forId = htmlForProp || FormUtils.findDescriptors(children)?.input?.[0]?.id;
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      Typography.HvTypography,
      {
        id: localId,
        className: cx(
          classes.root,
          {
            [classes.labelDisabled]: !!localDisabled,
            [classes.childGutter]: !!(children && label)
          },
          className
        ),
        variant: "label",
        component: "label",
        htmlFor: forId,
        ...others,
        children: [
          label,
          localRequired && /* @__PURE__ */ jsxRuntime.jsx("span", { "aria-hidden": "true", children: "*" })
        ]
      }
    ),
    children
  ] });
};
exports.labelClasses = Label_styles.staticClasses;
exports.HvLabel = HvLabel;
