// @hitachivantara/uikit-react-core/dist/esm/ListContainer/ListContainer.js
import { jsx } from "@emotion/react/jsx-runtime";
import React__default, { forwardRef, useRef, useContext, useMemo, isValidElement } from "react";
import { useForkRef } from "../hooks/useForkRef.js";
import { useDefaultProps } from "../hooks/useDefaultProps.js";
import { useClasses } from "./ListContainer.styles.js";
import { staticClasses } from "./ListContainer.styles.js";
import HvListContext from "./ListContext/ListContext.js";
const HvListContainer = forwardRef((props, ref) => {
  const {
    id,
    classes: classesProp,
    className,
    interactive = false,
    selectable,
    condensed,
    disableGutters,
    children: childrenProp,
    ...others
  } = useDefaultProps("HvListContainer", props);
  const { classes, cx } = useClasses(classesProp);
  const containerRef = useRef(null);
  const { topContainerRef, nesting = -1 } = useContext(HvListContext);
  const listContext = useMemo(
    () => ({
      topContainerRef: topContainerRef || containerRef,
      condensed,
      selectable,
      disableGutters,
      interactive,
      nesting: nesting + 1
    }),
    [
      condensed,
      selectable,
      disableGutters,
      interactive,
      nesting,
      topContainerRef
    ]
  );
  const children = useMemo(() => {
    if (!interactive)
      return childrenProp;
    const anySelected = React__default.Children.toArray(childrenProp).some(
      (child) => isValidElement(child) && child.props.selected && !child.props.disabled
    );
    return React__default.Children.map(childrenProp, (child, i) => {
      const tabIndex = child.props.tabIndex || !anySelected && i === 0 || child.props.selected && !child.props.disabled ? 0 : -1;
      return React__default.cloneElement(child, {
        tabIndex,
        interactive
      });
    });
  }, [childrenProp, interactive]);
  const handleRef = useForkRef(ref, containerRef);
  return /* @__PURE__ */ jsx(HvListContext.Provider, { value: listContext, children: /* @__PURE__ */ jsx(
    "ul",
    {
      ref: handleRef,
      id,
      className: cx(classes.root, className),
      ...others,
      children
    }
  ) });
});
export {
  HvListContainer,
  staticClasses as listContainerClasses
};
