// @hitachivantara/uikit-react-core/dist/esm/ScrollTo/Vertical/ScrollToVertical.js
import { jsx } from "@emotion/react/jsx-runtime";
import { useDefaultProps } from "../../hooks/useDefaultProps.js";
import { useUniqueId } from "../../hooks/useUniqueId.js";
import { isKey } from "../../utils/keyboardUtils.js";
import { setId } from "../../utils/setId.js";
import { useClasses, calculateOffset } from "./ScrollToVertical.styles.js";
import { staticClasses } from "./ScrollToVertical.styles.js";
import { useScrollTo } from "../useScrollTo.js";
import { HvVerticalScrollListItem } from "./VerticalScrollListItem/VerticalScrollListItem.js";
const HvScrollToVertical = (props) => {
  const {
    id,
    defaultSelectedIndex = 0,
    scrollElementId,
    href = true,
    navigationMode = href ? "push" : "none",
    relativeLinks = false,
    onChange,
    onClick,
    onEnter,
    className,
    classes: classesProp,
    options,
    offset = 0,
    position = "relative",
    tooltipPosition = "left",
    style,
    ...others
  } = useDefaultProps("HvScrollToVertical", props);
  const { classes, cx } = useClasses(classesProp);
  const elementId = useUniqueId(id, "hvVerticalScrollto");
  const [selectedIndex, setScrollTo, elements] = useScrollTo(
    defaultSelectedIndex,
    scrollElementId,
    navigationMode,
    relativeLinks,
    offset,
    options,
    onChange
  );
  const handleSelection = (event, value, index) => {
    event.preventDefault();
    const wrappedOnChange = () => {
      onChange?.(event, index);
    };
    setScrollTo(event, value, index, wrappedOnChange);
  };
  const tabs = elements.map((option, index) => /* @__PURE__ */ jsx(
    HvVerticalScrollListItem,
    {
      id: setId(elementId, `item-${index}`),
      onClick: (event) => {
        if (navigationMode !== "none") {
          event.preventDefault();
        }
        handleSelection(event, option.value, index);
        onClick?.(event, index);
      },
      onKeyDown: (event) => {
        if (isKey(event, "Enter") === true) {
          if (navigationMode !== "none") {
            event.preventDefault();
          }
          handleSelection(event, option.value, index);
          onEnter?.(event, index);
        }
      },
      href: navigationMode !== "none" ? option.href : void 0,
      tooltipPlacement: tooltipPosition,
      selected: selectedIndex === index,
      label: option.label
    },
    option.key || option.label
  ));
  const positionOffset = calculateOffset(options.length);
  return /* @__PURE__ */ jsx(
    "ol",
    {
      className: cx(
        classes.root,
        {
          [classes.positionFixed]: position === "fixed",
          [classes.positionAbsolute]: position === "absolute"
        },
        className
      ),
      style: { top: `calc(50% - ${positionOffset}px)`, ...style },
      id: elementId,
      ...others,
      children: tabs
    }
  );
};
export {
  HvScrollToVertical,
  staticClasses as scrollToVerticalClasses
};
