// @hitachivantara/uikit-react-core/dist/cjs/Header/Header.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../utils/classes.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvHeader", {
  root: {
    height: uikitStyles.theme.header.height,
    backgroundColor: uikitStyles.theme.colors.atmo1,
    display: "flex",
    flexDirection: "column",
    width: "100%",
    boxSizing: "border-box",
    flexShrink: 0,
    zIndex: uikitStyles.theme.zIndices.banner,
    color: uikitStyles.theme.colors.atmo1,
    boxShadow: uikitStyles.theme.colors.shadow,
    borderTop: "none"
  },
  header: {
    display: "flex",
    alignItems: "center",
    width: "100%",
    height: "100%",
    padding: `0 ${uikitStyles.theme.space.sm}`,
    boxShadow: uikitStyles.theme.colors.shadow,
    "& > *:not(nav)": {
      zIndex: 2
    }
  },
  backgroundColor: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
