// @hitachivantara/uikit-react-core/dist/cjs/TreeView/TreeItem/DefaultContent.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const useCss = require("../../hooks/useCss.cjs");
const classes = require("../../utils/classes.cjs");
const useHvTreeItem = require("./useHvTreeItem.cjs");
const { useClasses } = classes.createClasses("HvTreeContent", {
  root: {},
  expanded: {},
  selected: {},
  focused: {},
  disabled: {},
  iconContainer: {},
  label: {}
});
const DefaultContent = React.forwardRef(
  (props, ref) => {
    const {
      classes: classesProp,
      className,
      displayIcon,
      expansionIcon,
      icon: iconProp,
      label,
      nodeId,
      onClick,
      onMouseDown,
      ...others
    } = props;
    const { classes: classes2 } = useClasses(classesProp);
    const { cx } = useCss.useCss();
    const {
      disabled,
      expanded,
      selected,
      focused,
      handleExpansion,
      handleSelection,
      preventSelection
    } = useHvTreeItem.useHvTreeItem(nodeId);
    const icon = iconProp ?? expansionIcon ?? displayIcon;
    return (
      /* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions -- Key event is handled by the TreeView */
      /* @__PURE__ */ jsxRuntime.jsxs(
        "div",
        {
          ...others,
          className: cx(className, classes2.root, {
            [classes2.expanded]: expanded,
            [classes2.selected]: selected,
            [classes2.focused]: focused,
            [classes2.disabled]: disabled
          }),
          onClick: (event) => {
            handleExpansion(event);
            handleSelection(event);
            onClick?.(event);
          },
          onMouseDown: (event) => {
            preventSelection(event);
            onMouseDown?.(event);
          },
          ref,
          children: [
            /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes2.iconContainer, children: icon }),
            /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes2.label, children: label })
          ]
        }
      )
    );
  }
);
exports.DefaultContent = DefaultContent;
exports.useClasses = useClasses;
