// @hitachivantara/uikit-react-core/dist/cjs/Table/TableHead/TableHead.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const TableContext = require("../TableContext.cjs");
const TableSectionContext = require("../TableSectionContext.cjs");
const TableHead_styles = require("./TableHead.styles.cjs");
const tableSectionContext = {
  type: "head"
};
const defaultComponent = "thead";
const HvTableHead = React.forwardRef(
  ({ classes: classesProp, className, component, stickyHeader, ...others }, externalRef) => {
    const { classes, cx } = TableHead_styles.useClasses(classesProp);
    const tableContext = React.useContext(TableContext.default);
    const Component = component || tableContext?.components?.THead || defaultComponent;
    return /* @__PURE__ */ jsxRuntime.jsx(TableSectionContext.default.Provider, { value: tableSectionContext, children: /* @__PURE__ */ jsxRuntime.jsx(
      Component,
      {
        className: cx(
          classes.root,
          { [classes.stickyHeader]: stickyHeader },
          className
        ),
        ref: externalRef,
        role: Component === defaultComponent ? null : "rowgroup",
        ...others
      }
    ) });
  }
);
exports.tableHeadClasses = TableHead_styles.staticClasses;
exports.HvTableHead = HvTableHead;
