// @hitachivantara/uikit-react-core/dist/cjs/FileUploader/DropZone/DropZone.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const uniqueId = require("lodash/uniqueId");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const setId = require("../../utils/setId.cjs");
const useUniqueId = require("../../hooks/useUniqueId.cjs");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const utils = require("../utils.cjs");
const DropZone_styles = require("./DropZone.styles.cjs");
const Label = require("../../Forms/Label/Label.cjs");
const InfoMessage = require("../../Forms/InfoMessage/InfoMessage.cjs");
const Typography = require("../../Typography/Typography.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const uniqueId__default = /* @__PURE__ */ _interopDefault(uniqueId);
function validateAccept(file, acceptAttr) {
  if (!file || !acceptAttr)
    return true;
  const acceptEntries = acceptAttr.split(",");
  const fileName = file.name || "";
  const mimeType = (file.type || "").toLowerCase();
  const baseMimeType = mimeType.replace(/\/.*$/, "");
  return acceptEntries.some((type) => {
    const validType = type.trim().toLowerCase();
    if (validType.charAt(0) === ".") {
      return fileName.toLowerCase().endsWith(validType);
    }
    if (validType.endsWith("/*")) {
      return baseMimeType === validType.replace(/\/.*$/, "");
    }
    return mimeType === validType;
  });
}
const HvDropZone = (props) => {
  const {
    id: idProp,
    classes: classesProp,
    labels,
    accept,
    maxFileSize,
    inputProps,
    hideLabels,
    multiple = true,
    disabled = false,
    onFilesAdded
  } = useDefaultProps.useDefaultProps("HvDropZone", props);
  const id = useUniqueId.useUniqueId(idProp, "dropzone");
  const { classes, cx } = DropZone_styles.useClasses(classesProp);
  const [dragState, setDragState] = React.useState(false);
  const inputRef = React.useRef(null);
  const handleDragLeave = () => {
    setDragState(false);
  };
  const handleDragEnter = (event) => {
    if (disabled)
      return;
    event.stopPropagation();
    event.preventDefault();
    setDragState(true);
  };
  const onChangeHandler = (filesList) => {
    const filesToProcess = Object.values(filesList);
    const newFiles = filesToProcess.map((file) => {
      const newFile = new File([file], file.name, {
        type: file.type,
        lastModified: file.lastModified
      });
      newFile.id = uniqueId__default.default("uploaded-file-data-");
      const isSizeAllowed = file.size <= maxFileSize;
      const isFileAccepted = !accept || accept.includes(file.type?.split("/")[1]) || // TODO: remove in v6
      validateAccept(file, accept);
      if (!isFileAccepted) {
        newFile.errorMessage = labels?.fileTypeError;
        newFile.status = "fail";
      } else if (!isSizeAllowed) {
        newFile.errorMessage = labels?.fileSizeError;
        newFile.status = "fail";
      }
      return newFile;
    });
    onFilesAdded?.(newFiles);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    !hideLabels && /* @__PURE__ */ jsxRuntime.jsxs("div", { id, className: classes.dropZoneLabelsGroup, children: [
      /* @__PURE__ */ jsxRuntime.jsx(
        Label.HvLabel,
        {
          id: setId.setId(id, "input-file-label"),
          htmlFor: setId.setId(id, "input-file"),
          label: labels?.dropzone,
          className: classes.dropZoneLabel
        }
      ),
      /* @__PURE__ */ jsxRuntime.jsxs(InfoMessage.HvInfoMessage, { id: setId.setId(id, "description"), children: [
        Number.isInteger(maxFileSize) && `${labels?.sizeWarning} ${utils.convertUnits(maxFileSize)}`,
        labels?.acceptedFiles ? labels.acceptedFiles : accept && ` (${accept?.replaceAll(",", ", ")})`
      ] })
    ] }),
    /* @__PURE__ */ jsxRuntime.jsxs(
      "div",
      {
        id: setId.setId(id, "input-file-container"),
        className: cx(classes.dropZoneContainer, {
          [classes.dragAction]: dragState,
          [classes.dropZoneContainerDisabled]: disabled
        }),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(
            "input",
            {
              id: setId.setId(id, "input-file"),
              className: classes.inputArea,
              type: "file",
              multiple,
              disabled,
              title: !disabled ? `${labels?.drag} ${labels?.selectFiles}` : "",
              onClick: () => {
                if (inputRef.current) {
                  inputRef.current.value = "";
                }
              },
              onChange: () => {
                if (!disabled && inputRef.current?.files) {
                  onChangeHandler(inputRef.current.files);
                }
              },
              onDragEnter: handleDragEnter,
              onDragOver: handleDragEnter,
              onDragLeave: handleDragLeave,
              onDropCapture: handleDragLeave,
              onDrop: (event) => {
                if (disabled)
                  return;
                const { files } = event.dataTransfer;
                if (multiple === true || files.length === 1) {
                  event.stopPropagation();
                  event.preventDefault();
                  onChangeHandler(files);
                }
              },
              ref: inputRef,
              accept,
              ...inputProps
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes?.dropArea, children: dragState ? /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.dropZoneAreaLabels, children: /* @__PURE__ */ jsxRuntime.jsx(Typography.HvTypography, { className: classes.dragText, children: labels?.dropFiles }) }) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
            /* @__PURE__ */ jsxRuntime.jsx(
              uikitReactIcons.Doc,
              {
                iconSize: "M",
                className: classes.dropZoneAreaIcon,
                color: disabled ? "secondary_60" : "secondary"
              }
            ),
            /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.dropZoneAreaLabels, children: /* @__PURE__ */ jsxRuntime.jsxs(Typography.HvTypography, { className: classes.dragText, children: [
              labels?.drag,
              /* @__PURE__ */ jsxRuntime.jsx(
                "span",
                {
                  className: classes.selectFilesText,
                  children: ` ${labels?.selectFiles}`
                }
              )
            ] }) })
          ] }) })
        ]
      }
    )
  ] });
};
exports.dropZoneClasses = DropZone_styles.staticClasses;
exports.HvDropZone = HvDropZone;
