// @hitachivantara/uikit-react-core/dist/cjs/ScrollTo/Vertical/VerticalScrollListItem/VerticalScrollListItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../utils/focusUtils.cjs");
const classes = require("../../../utils/classes.cjs");
const name = "HvVerticalScrollListItem";
const { staticClasses, useClasses } = classes.createClasses(name, {
  root: {
    padding: "0",
    height: "32px",
    width: "32px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  },
  notSelected: {
    height: "4px",
    width: "4px",
    borderRadius: "50%",
    display: "inline-block",
    backgroundColor: uikitStyles.theme.colors.secondary_60
  },
  // TODO: remove in v6 (use classes.button)
  text: {},
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "16px",
    width: "16px",
    borderRadius: "50%",
    cursor: "pointer",
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover,
      "& $notSelected": {
        height: "6px",
        width: "6px",
        backgroundColor: uikitStyles.theme.colors.secondary
      }
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
