// @hitachivantara/uikit-react-core/dist/esm/AvatarGroup/AvatarGroup.js
import { jsxs, jsx } from "@emotion/react/jsx-runtime";
import { forwardRef, Children, cloneElement } from "react";
import { css } from "@emotion/css";
import { theme } from "@hitachivantara/uikit-styles";
import { useDefaultProps } from "../hooks/useDefaultProps.js";
import { useClasses } from "./AvatarGroup.styles.js";
import { staticClasses } from "./AvatarGroup.styles.js";
import { HvAvatar } from "../Avatar/Avatar.js";
const getSpacingValue = (spacing, size) => {
  switch (size) {
    case "xs":
      return spacing === "compact" ? 24 : 16;
    case "sm":
      return spacing === "compact" ? 30 : 18;
    case "md":
      return spacing === "compact" ? 36 : 20;
    case "lg":
      return spacing === "compact" ? 44 : 24;
    case "xl":
      return spacing === "compact" ? 72 : 34;
    default:
      return spacing === "compact" ? 30 : 18;
  }
};
const HvAvatarGroup = forwardRef(
  (props, ref) => {
    const {
      className,
      classes: classesProp,
      children,
      size = "sm",
      spacing = "loose",
      direction = "row",
      toBack = true,
      maxVisible = 3,
      overflowComponent,
      ...others
    } = useDefaultProps("HvAvatarGroup", props);
    const { classes, cx } = useClasses(classesProp);
    const spacingValue = getSpacingValue(spacing, size);
    const totalChildren = Children.count(children);
    const zIndexMultiplier = toBack ? -1 : 1;
    const willOverflow = totalChildren > maxVisible;
    return /* @__PURE__ */ jsxs(
      "div",
      {
        className: cx(classes.root, classes[direction], className),
        ref,
        ...others,
        children: [
          Children.map(children, (child, index) => {
            if (index < maxVisible) {
              return cloneElement(child, {
                style: {
                  zIndex: 100 + index * zIndexMultiplier
                },
                classes: {
                  container: css({
                    marginLeft: direction === "row" ? index !== 0 ? -spacingValue : 0 : 0,
                    marginTop: direction === "column" ? index !== 0 ? -spacingValue : 0 : 0
                  })
                },
                size
              });
            }
          }),
          willOverflow && /* @__PURE__ */ jsx(
            "div",
            {
              style: {
                marginLeft: direction === "row" ? -spacingValue : 0,
                marginTop: direction === "column" ? -spacingValue : 0,
                zIndex: 100 + maxVisible * zIndexMultiplier
              },
              children: overflowComponent ? overflowComponent(totalChildren - maxVisible) : /* @__PURE__ */ jsxs(
                HvAvatar,
                {
                  size,
                  backgroundColor: theme.colors.atmo4,
                  classes: {
                    avatar: css({
                      [`&.HvAvatar-${size}`]: {
                        fontSize: "unset"
                      }
                    })
                  },
                  children: [
                    "+",
                    totalChildren - maxVisible
                  ]
                }
              )
            }
          )
        ]
      }
    );
  }
);
export {
  HvAvatarGroup,
  staticClasses as avatarGroupClasses
};
