// @hitachivantara/uikit-react-core/dist/cjs/Dropdown/Dropdown.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../utils/classes.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvDropdown", {
  root: {
    width: "100%",
    position: "relative",
    display: "inline-block"
  },
  labelContainer: { display: "flex", alignItems: "flex-start" },
  label: { paddingBottom: "6px", display: "block" },
  description: {},
  error: {},
  placeholder: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  },
  placeholderClosed: {
    color: uikitStyles.theme.colors.secondary_80
  },
  selectionDisabled: {
    lineHeight: uikitStyles.theme.space.md,
    color: uikitStyles.theme.colors.secondary_60
  },
  dropdown: {
    width: "100%",
    borderRadius: uikitStyles.theme.radii.base,
    "& $dropdownHeaderInvalid": {
      border: `1px solid ${uikitStyles.theme.colors.negative}`
    }
  },
  arrow: {},
  dropdownHeader: {},
  dropdownHeaderInvalid: {},
  dropdownHeaderOpen: {},
  dropdownListContainer: {},
  rootList: {},
  disabled: {
    color: uikitStyles.theme.colors.secondary_60
  },
  readOnly: {
    "& $dropdownHeader": {
      border: `1px solid ${uikitStyles.theme.colors.secondary_80}`,
      backgroundColor: uikitStyles.theme.colors.atmo1
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
