// @hitachivantara/uikit-react-core/dist/cjs/Section/Section.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../utils/classes.cjs");
const { staticClasses, useClasses } = classes.createClasses("HvSection", {
  root: {
    width: "100%",
    display: "flex",
    flexDirection: "column",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    borderRadius: uikitStyles.theme.radii.round,
    border: `1px solid ${uikitStyles.theme.colors.atmo4}`
  },
  hidden: { height: 0, display: "none" },
  header: {
    display: "flex",
    alignItems: "center",
    position: "relative",
    minHeight: uikitStyles.theme.sizes.sm,
    padding: uikitStyles.theme.space.sm
  },
  content: {
    padding: uikitStyles.theme.spacing(0, "sm", "sm", "sm")
  },
  spaceTop: {
    paddingTop: uikitStyles.theme.space.sm
  },
  actions: {
    display: "flex",
    gap: uikitStyles.theme.space.xs,
    position: "absolute",
    right: 0
  },
  raisedHeader: {
    zIndex: 1,
    boxShadow: uikitStyles.theme.colors.shadow,
    "+ div": {
      paddingTop: uikitStyles.theme.space.sm
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
