// @hitachivantara/uikit-react-core/dist/cjs/Card/Header/Header.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const MuiCardHeader = require("@mui/material/CardHeader");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const Header_styles = require("./Header.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const MuiCardHeader__default = /* @__PURE__ */ _interopDefault(MuiCardHeader);
const HvCardHeader = (props) => {
  const {
    classes: classesProp,
    className,
    title,
    subheader,
    icon,
    onClick,
    ...others
  } = useDefaultProps.useDefaultProps("HvCardHeader", props);
  const { classes, cx } = Header_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    MuiCardHeader__default.default,
    {
      title,
      subheader,
      action: icon,
      onClick,
      className: cx(classes.root, className),
      classes: {
        title: cx({
          // TODO: review conditional in v6
          [classes.titleShort]: icon,
          [classes.title]: !icon
        }),
        subheader: classes.subheader,
        action: classes.action,
        content: classes.content
      },
      ...others
    }
  );
};
exports.cardHeaderClasses = Header_styles.staticClasses;
exports.HvCardHeader = HvCardHeader;
