// @hitachivantara/uikit-react-core/dist/cjs/Header/Navigation/MenuItem/MenuItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../utils/focusUtils.cjs");
const classes = require("../../../utils/classes.cjs");
const item = {
  border: "none",
  cursor: "pointer",
  padding: `${uikitStyles.theme.space.xs} ${uikitStyles.theme.space.sm}`,
  height: "100%",
  display: "flex",
  alignItems: "center",
  "&:active": {
    outline: "none"
  },
  "&:focus": {
    outline: "none"
  },
  "&:focus-visible": {
    ...focusUtils.outlineStyles
  },
  "& span": {
    display: "inline-flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-between"
  },
  "& span::after": {
    content: "attr(data-text)",
    height: 0,
    visibility: "hidden",
    overflow: "hidden",
    userSelect: "none",
    pointerEvents: "none",
    fontWeight: uikitStyles.theme.fontWeights.semibold,
    "@media speech": {
      display: "none"
    }
  }
};
const { staticClasses, useClasses } = classes.createClasses(
  "HvHeader-MenuItem",
  {
    root: {
      display: "inline",
      "&:hover": {
        backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
      },
      "&:focus-within": {
        backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
      }
    },
    menu: {
      marginTop: 0,
      paddingBottom: "4px",
      paddingTop: "4px",
      height: "100%",
      borderTop: "none",
      borderBottom: `4px solid ${uikitStyles.theme.colors.atmo2}`
    },
    menubar: {
      marginTop: "0px",
      paddingBottom: "4px",
      paddingTop: "4px",
      height: "100%",
      borderTop: `0px solid ${uikitStyles.theme.colors.atmo1}`,
      borderBottom: `${"4px"} solid ${uikitStyles.theme.colors.atmo1}`
    },
    selected: {
      "&$menu": {
        marginTop: 0,
        paddingBottom: 0,
        borderTop: "none",
        borderBottom: `4px solid ${uikitStyles.theme.colors.secondary}`,
        paddingTop: "4px",
        height: "100%",
        "& > * > span": {
          color: uikitStyles.theme.colors.secondary
        }
      },
      "&$menubar": {
        marginTop: 0,
        paddingBottom: 0,
        borderTop: "none",
        borderBottom: `4px solid ${uikitStyles.theme.colors.secondary}`,
        paddingTop: "4px",
        height: "100%"
      }
    },
    link: {
      textDecoration: "none",
      ...item
    },
    button: {
      color: "inherit",
      ...item
    }
  }
);
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
